### PR TITLE
Issue #1425: Fix token view modes

### LINKED ISSUE #1425: Token display format is not configurable (and date token creates very long paths)
I've created a date field, and want to include the start date of an event in its url alias pattern:
events/[node:field_event_date]-[node:title]

However, when the path is generated, it uses the complete start and end dates and times, like:
http://test-datera.pantheon.io/events/2015-12-18-220000-2015-12-18-230000-second-website-demo

I don't remember having trouble adding the simple start date when doing this in D7. How are we handling things differently, now that we've moved the token module to Backdrop core?

---

~PR by @docwilmot: https://github.com/backdrop/backdrop/pull/2476~
PR by @docwilmot: https://github.com/backdrop/backdrop/pull/3101

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/1425

### CODE REVIEW

File: core/modules/entity/entity.module [hunk 1]
```diff
@@ -18,6 +18,29 @@ function entity_modules_disabled() {
   entity_info_cache_clear();
 }
 
+/**
+ * Implements hook_entity_info_alter().
+ *
+ * @since version 1.12.0
```
Comment (klonos) [inline core/modules/entity/entity.module]: We should update this to either 1.13.4, or 1.14.0

File: core/modules/field/field.tokens.inc [hunk 2]
```diff
@@ -48,6 +48,54 @@ function field_tokens($type, $tokens, array $data = array(), array $options = ar
...
   $langcode = isset($options['language']) ? $options['language']->langcode : NULL;
 
+  // Field value tokens.
+  if ($type == 'field-value' && !empty($data['field'])) {
+    $field = $data['field'];
+    $type_info = token_get_info($field['field_token_type']);
+    $field_value_token_type = $type_info['field-value-type'];
+
+    foreach ($tokens as $name => $original) {
+      switch ($name) {
```
Comment (klonos) [inline core/modules/field/field.tokens.inc]: Why use `switch` here, if there's only one `case`?

**Comment by herbdool** [comment]
New sandbox

**Comment by quicksketch** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/3101.

**Pull request closed without merging**